packages/flutter/lib/src/material/ink_well.dart

```diff
@@ -197,14 +197,10 @@ class _InkResponseState<T extends InkResponse> extends State<T> {
   }
 
   @override
-  void dependenciesChanged(Type affectedWidgetType) {
-    if (affectedWidgetType == Theme && _lastHighlight != null)
-      _lastHighlight.color = Theme.of(context).highlightColor;
-  }
-
-  @override
   Widget build(BuildContext context) {
     assert(config.debugCheckContext(context));
+    if (_lastHighlight != null)
+      _lastHighlight.color = Theme.of(context).highlightColor;
```

Comment:
Yup, this works. (I tested that this doesn't call the RHS if the LHS evaluates to null.)
Done.